cmd/kubeadm/app/apis/kubeadm/v1alpha1/types.go

```diff
@@ -48,6 +48,16 @@ type MasterConfiguration struct {
 	APIServerCertSANs []string `json:"apiServerCertSANs"`
 	// CertificatesDir specifies where to store or look for all required certificates
 	CertificatesDir string `json:"certificatesDir"`
+
+	MasterCertificates *MasterCertificates `json:"masterCertificates,omitempty"`
+	// Public Addr or DNS
+	PublicAddress string `json:"publicAddress"`
+	//Masters Count
+	Count int `json:"count"`
+
+	//required for cloud providers. To make right tag for resources
+	ClusterName      string
```

Comment:
re: discussion there are a couple of items: 
meta link is here:  https://github.com/kubernetes/kubeadm/issues/261 which also references the doc.